ui/app/AppLayouts/Chat/stores/MessageStore.qml

```diff
@@ -225,6 +227,12 @@ QtObject {
         messageModule.leaveChat();
     }
 
+    function addNewMessagesMarker() {
```

Comment:
Having QML tell Nim to add a marker seems weird. The backend should know when to do that already.

Reply:
There is one case when the backend doesn't know when/if to add a new messages marker. See the video attached to the description 0:48->0:56. In order to do that backend would need to know if the most recent message is visible in the viewport.

There are two options:
1) **qml**: when(hasUnreadMessagesChanged) and if (not mostRecentMesssageInViewport) then (add marker)--->**view.nim**
2) **view.nim**: when(hasUnreadMessagesChanged) and if (not mostRecentMesssageInViewport)--->**qml** add marker

In the second case `view.nim` would need to know about the qml state. It doesn't seem right. That's why I decided to go with option 1.

Reply:
Ok makes sense!